webapp/src/components/modals/create_issue/index.tsx

```diff
@@ -0,0 +1,284 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {Modal} from 'react-bootstrap';
+import {Theme} from 'mattermost-redux/types/preferences';
+import {getPost} from 'mattermost-redux/selectors/entities/posts';
+
+import {getErrorMessage} from 'src/utils/user_utils';
+import GitlabLabelSelector from 'src/components/gitlab_label_selector';
+import GitlabAssigneeSelector from 'src/components/gitlab_assignee_selector';
+import GitlabMilestoneSelector from 'src/components/gitlab_milestone_selector';
+import GitlabProjectSelector from 'src/components/gitlab_project_selector';
+import Validator from 'src/components/validator';
+import FormButton from 'src/components/form_button';
+import Input from 'src/components/input';
+import {id as pluginId} from 'src/manifest';
+import {closeCreateIssueModal, createIssue} from 'src/actions';
+import {GlobalState} from 'src/types/global_state';
+import {usePrevious} from 'src/hooks/use_previous';
+
+const MAX_TITLE_LENGTH = 256;
+
+type PropTypes = {
+    theme: Theme;
+};
+
+const CreateIssueModal = (props: PropTypes) => {
+    const [validator, setValidator] = useState(new Validator());
+    const [submitting, setSubmitting] = useState<boolean>(false);
+    const [error, setError] = useState<string | null>(null);
+    const [project, setProject] = useState<ProjectSelection | null>(null);
+    const [issueTitle, setIssueTitle] = useState<string>('');
+    const [issueDescription, setIssueDescription] = useState<string>('');
+    const [labels, setLabels] = useState<SelectionType[]>([]);
+    const [assignees, setAssignees] = useState<SelectionType[]>([]);
+    const [milestone, setMilestone] = useState<SelectionType | null>(null);
+    const [showErrors, setShowErrors] = useState<boolean>(false);
+    const [issueTitleValid, setIssueTitleValid] = useState<boolean>(true);
+
+    const {visible, post, channelId, title} = useSelector((state: GlobalState) => {
+        const {postId, title, channelId} = state[`plugins-${pluginId}` as plugin].createIssueModal;
+        
+        const post = postId ? getPost(state, postId) : null;
+        return {
+            visible: state[`plugins-${pluginId}` as plugin].isCreateIssueModalVisible,
+            post,
+            title,
+            channelId,
+        };
+    });
+
+    const dispatch = useDispatch();
+
+    const prevPost = usePrevious(post);
+    const prevChannelId = usePrevious(channelId);
+    const prevTitle = usePrevious(title);
+
+    useEffect(() => {        
+        if (post && !prevPost) {
+            setIssueDescription(post.message);
+        } else if (channelId && (channelId !== prevChannelId || title !== prevTitle)) {
+            setIssueTitle(title.substring(0, MAX_TITLE_LENGTH));
+        }
+    }, [props, prevPost, prevChannelId, prevTitle, channelId, title, post]);
```

Comment:
```suggestion
    }, [channelId, title, post]);
```
Is adding the `prev` fields really needed here? Their value changes every time the value of the original field changes.

And what's the point of mentioning the `props` here. Those only contain the theme. Should we run the function whenever the theme changes?